internal/util/parse/parse.go

```diff
@@ -22,113 +22,18 @@ import (
 	"path/filepath"
 	"strings"
 
-	kpterrors "github.com/GoogleContainerTools/kpt/internal/errors"
 	"github.com/GoogleContainerTools/kpt/internal/gitutil"
 	kptfilev1 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1"
-	"github.com/google/go-containerregistry/pkg/name"
 	"sigs.k8s.io/kustomize/kyaml/errors"
 )
 
-type Options struct {
-	SetGit func(git *kptfilev1.Git) error
-	SetOci func(oci *kptfilev1.Oci) error
-}
-
-func ParseArgs(ctx context.Context, args []string, opts Options) (string, error) {
-	const op kpterrors.Op = "parse.ParseArgs"
-
-	tGit, errGit := GitParseArgs(ctx, args)
-	if errGit == nil {
-		if opts.SetGit == nil {
-			return "", kpterrors.E(op, fmt.Errorf("git locations not supported: %v", errGit))
-		}
-		if err := opts.SetGit(&tGit.Git); err != nil {
-			return "", err
-		}
-		return tGit.Destination, nil
-	}
-
-	tOci, errOci := OciParseArgs(ctx, args)
-	if errOci == nil {
-		if opts.SetOci == nil {
-			return "", kpterrors.E(op, fmt.Errorf("oci locations not supported: %v", errOci))
-		}
-		if err := opts.SetOci(&tOci.Oci); err != nil {
-			return "", err
-		}
-		return tOci.Destination, nil
-	}
-
-	// TODO(oci-support) combining error messages like this is suboptimal in several ways
-	return "", kpterrors.E(op, fmt.Errorf("%v %v", errGit, errOci))
-}
-
-type OciTarget struct {
-	kptfilev1.Oci
-	Destination string
-}
-
-func OciParseArgs(ctx context.Context, args []string) (OciTarget, error) {
-	oci := OciTarget{}
-	if args[0] == "-" {
-		return oci, nil
-	}
-
-	// The prefix must occur, and must not have other characters before it
-	arg0parts := strings.SplitN(args[0], "oci://", 2)
-	if len(arg0parts) != 2 || len(arg0parts[0]) != 0 {
-		return oci, errors.Errorf("ambiguous image:tag specify 'oci://' before argument: %s", args[0])
-	}
-
-	return targetFromImageReference(arg0parts[1], args[1])
-}
-
-func targetFromImageReference(image, dest string) (OciTarget, error) {
-	ref, err := name.ParseReference(image)
-	if err != nil {
-		return OciTarget{}, err
-	}
-
-	registry := ref.Context().RegistryStr()
-	repository := ref.Context().RepositoryStr()
-	destination, err := getDest(dest, registry, repository)
-	if err != nil {
-		return OciTarget{}, err
-	}
-
-	directory := ""
-	parts := strings.SplitN(ref.Context().Name(), "//", 2)
-	if len(parts) == 2 {
-		directory = "/" + parts[1]
-		repo, err := name.NewRepository(parts[0])
-		if err != nil {
-			return OciTarget{}, err
-		}
-
-		switch r := ref.(type) {
-		case name.Tag:
-			ref = repo.Tag(r.TagStr())
-		case name.Digest:
-			ref = repo.Tag(r.DigestStr())
-		}
-	}
-
-	return OciTarget{
-		Oci: kptfilev1.Oci{
-			Image:     ref.Name(),
-			Directory: directory,
-		},
-		Destination: destination,
-	}, nil
-}
-
-type GitTarget struct {
+type Target struct {
 	kptfilev1.Git
 	Destination string
 }
 
-func GitParseArgs(ctx context.Context, args []string) (GitTarget, error) {
-	g := GitTarget{}
+func GitParseArgs(ctx context.Context, args []string) (Target, error) {
```

Comment:
Same note - this file is good to spot check. Actually - rather than wait for PR to main - you could spot-check now with a copy-paste of the entire raw file from main in your working copy to verify that it remains "unchanged".

Reply:
ack. Will double check this file too

Reply:
confirmed. they are near identical; porch contains the filesystem changes which makes them not exactly identical.